### ISSUE TITLE
Add an option to allow adding metadata (json) in upload

### ISSUE THREAD

**Issue by arnou** [dadoonet/fscrawler#335]
It would be great to be able to add on-the-fly contextual metadata with the upload API: so you could index contextual data for subsequent queries (using for example a new metadata form field)

Dominique

**Comment by arnou** [comment]
But if possible, we should also have the same functionality with the synchronization job. For example, if a file namefile.pdf_meta.json is found in the same directory as the main file namefile.pdf, its contents are considered as metadata

**Comment by dadoonet** [comment]
> we should also have the same functionality with the synchronization job. For example, if a file namefile.pdf_meta.json is found in the same directory as the main file namefile.pdf, its contents are considered as metadata

Do you mean that we would have to upload 2 files or something along those lines?

> It would be great to be able to add on-the-fly contextual metadata with the upload API: so you could index contextual data for subsequent queries (using for example a new metadata form field)

That could make sense indeed.

**Comment by arnou** [comment]
Yes, it's two files I thought, but we could probably imagine a single file per directory also .. it should also impact your ticket #334

**Comment by dadoonet** [comment]
But you mentioned `upload` in your title so I assume you want to do that over the REST API. Do I understand correctly?

If so, what is the point of `a single file per directory also`? We don't upload directories over REST.

**Comment by arnou** [comment]
Sorry, I was not clear enough.

The ticket mainly concerns the upload functionality of the REST API.

For this purpose alone, the metadata can be added to the API via a form field, either text or file type

Then, when I added the second comment (this should perhaps be the subject of another ticket), I was thinking of matching as much as possible the functionality offered by the synchronization of the file system and those of the " API REST (if I can index a file and external metadata to it with the API, I should be able to do the same with the file system)

Tell me if that clarifies things.

**Comment by dadoonet** [comment]
I see. Thanks for clarifying.

**Comment by dadoonet** [comment]
Will be fixed by #564

**Comment by dadoonet** [comment]
I think this is now available with #564.
